### PR TITLE
feat(wavefunction): accelerate sampling

### PR DESCRIPTION
**Pull request by AthenaCaesura**
## Description

Smartly pick when to format the samples obtained from wavefunctions.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [ ] I have updated documentation.

### CODE REVIEW

File: tests/orquestra/quantum/wavefunction_test.py
Comment (mstechly) [inline tests/orquestra/quantum/wavefunction_test.py:380]: This test shouldn't:
a) be here
b) look like this :)
Please see the [performance tests repo](https://github.com/zapatacomputing/orquestra-performance-tests) :)

**Review by mstechly — CHANGES_REQUESTED** [review]

File: tests/orquestra/quantum/wavefunction_test.py
Comment (AthenaCaesura) [reply tests/orquestra/quantum/wavefunction_test.py]: Got rid of it!

**Review by mstechly — COMMENTED** [review]
Thanks for getting rid of that!
Not approving as waiting for review from @dexter2206

**Review by dexter2206 — APPROVED** [review]

**Review by mstechly — APPROVED** [review]

**Pull request merged**
